plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsRepository.java

```diff
@@ -70,17 +73,18 @@ public HdfsRepository(
             throw new IllegalArgumentException("No 'uri' defined for hdfs snapshot/restore");
         }
         uri = URI.create(uriSetting);
-        if ("hdfs".equalsIgnoreCase(uri.getScheme()) == false) {
+        if (uri.getScheme() == null || allowedSchemes.contains(uri.getScheme().toLowerCase(Locale.ROOT)) == false) {
```

Comment:
I've removed the `toLowerCase` call now, but I want to highlight that it was using `equalsIgnoreCase` for comparison before my change.

So, I guess that technically this could be a breaking change for someone, however unlikely. Normally, it will just throw an exception if you try to use uppercased `HDFS` as a scheme:

```
org.apache.hadoop.fs.UnsupportedFileSystemException: fs.AbstractFileSystem.HDFS.impl=null: No AbstractFileSystem configured for scheme: HDFS
```

However, technically, you can avoid that error by setting `conf.fs.AbstractFileSystem.HDFS.impl` to `org.apache.hadoop.hdfs.DistributedFileSystem`, or maybe even set that to a different, custom implementation class.

So, that's why I just proposed the `toLowerCase` call initially - I wanted to avoid breaking some edge-case setup. Would this need to be a deprecation first?